torchvision/prototype/transforms/functional/_dispatch.py

```diff
@@ -0,0 +1,448 @@
+#
+# DO NOT CHANGE MANUALLY!
+
+from typing import Any, TypeVar, List, Optional, Tuple
+
+import torch
+import torchvision.prototype.transforms.functional as F
+import torchvision.transforms.functional as _F
+from torchvision.prototype import features
+from torchvision.prototype.features import BoundingBoxFormat, ColorSpace
+from torchvision.transforms import InterpolationMode
+
+Dispatcher = F.utils.Dispatcher
+
+# This is just a sentinel to have a default argument for a dispatcher if the feature specific implementations use
+# different defaults. The actual value is never used.
+FEATURE_SPECIFIC_DEFAULT = object()
+
+T = TypeVar("T", bound=features.Feature)
+
+
+__all__ = [
+    "horizontal_flip",
+    "resize",
+    "center_crop",
+    "normalize",
+    "resized_crop",
+    "erase",
+    "mixup",
+    "cutmix",
+    "affine",
+    "rotate",
+    "adjust_brightness",
+    "adjust_saturation",
+    "adjust_contrast",
+    "adjust_sharpness",
+    "posterize",
+    "solarize",
+    "autocontrast",
+    "equalize",
+    "invert",
+]
+
+
+@Dispatcher
+def horizontal_flip(input: T) -> T:
+    """ADDME"""
+    pass
+
+
+@horizontal_flip.implements(features.Image)
+def _horizontal_flip_image(input: features.Image) -> features.Image:
+    output = F.horizontal_flip_image(input)
+
+    return features.Image.new_like(input, output)
+
+
+@horizontal_flip.implements(features.BoundingBox)
+def _horizontal_flip_bounding_box(input: features.BoundingBox) -> features.BoundingBox:
+    intermediate_format = BoundingBoxFormat.XYXY
+    converted_input = F.convert_bounding_box_format(input, old_format=input.format, new_format=intermediate_format)
+
+    output = F.horizontal_flip_bounding_box(converted_input, image_size=input.image_size)
+    output = F.convert_bounding_box_format(output, old_format=intermediate_format, new_format=input.format)
+
+    return features.BoundingBox.new_like(input, output)
+
+
+@Dispatcher
+def resize(
+    input: T,
+    *,
+    size: List[int],
+    interpolation: InterpolationMode = FEATURE_SPECIFIC_DEFAULT,  # type: ignore[assignment]
+    max_size: Optional[int] = None,
+    antialias: Optional[bool] = None,
+) -> T:
+    """ADDME"""
+    pass
+
+
+@resize.implements(features.Image, pil_kernel=_F.resize)
+def _resize_image(
+    input: features.Image,
+    *,
+    size: List[int],
+    interpolation: InterpolationMode = InterpolationMode.BILINEAR,
+    max_size: Optional[int] = None,
+    antialias: Optional[bool] = None,
+) -> features.Image:
+    output = F.resize_image(input, size=size, interpolation=interpolation, max_size=max_size, antialias=antialias)
+
+    return features.Image.new_like(input, output)
+
+
+@resize.implements(features.BoundingBox)
+def _resize_bounding_box(input: features.BoundingBox, *, size: List[int], **_: Any) -> features.BoundingBox:
+    intermediate_format = BoundingBoxFormat.XYXY
+    converted_input = F.convert_bounding_box_format(input, old_format=input.format, new_format=intermediate_format)
```

Comment:
`resize_bounding_box` makes the assumption that the input is always in the XYXY format since a vanilla tensor doesn't carry this information. Since we have a `BoundingBox` feature here, we know what format the input has, we can convert it to the correct one. 

We can reasonably argue that the conversion could also happen at kernel level. But keep in mind that this means that a user has to pass `format=...` to every bounding box kernel that needs this conversion. Given that users will most likely only call the kernels together with JIT scripting or other low-level applications, I'm ok with both options.

Reply:
The benefit of moving this to the kernel level is that kernels are completely self-contained with the entire logic they need. Also adding the code there means that JIT users will be able to access the entire logic of the kernel which is great. Another benefit is that we keep the dispatch code VERY minimal without any custom logic. This could allow us to completely eliminate method bodies as discussed above.

The only drawback of pushing the entire code in the kernels is that for images, where we will have to support PIL, we would need to have that extra code on the kernel (the verification mostly). But from what I can see so far, that's a reasonable compromise for keeping this layer as thin as possible.

If you see other drawbacks it's worth discussing them. Otherwise I would be OK moving this to the kernel.

Reply:
See https://github.com/pytorch/vision/pull/5323#pullrequestreview-874795726.